lib/spack/spack/cmd/config.py

```diff
@@ -215,6 +243,33 @@ def config_list(args):
     print(" ".join(list(spack.config.SECTION_SCHEMAS)))
 
 
+def _config_scope_info_string(args, scope):
+    if args.show_paths and hasattr(scope, "path"):
+        section_path = scope.get_section_filename(args.section) if args.section else None
+        path = section_path if section_path and os.path.exists(section_path) else f"{scope.path}/"
+        return f"{scope.name} ({path})"
+    else:
+        return scope.name
+
+
+def config_scopes(args):
+    """List configured scopes in descending order of precedence."""
+
+    if args.section and not args.show_paths:
+        tty.warning(f"[section] ({args.section}) ignored without --show-path")
```

Comment:
Unfortunately, `tty` uses `warn` instead of `warning`:

```suggestion
        tty.warn(f"[section] ({args.section}) ignored without --show-paths")
```

Reply:
Would also like to see the shortcut (`-s`) here.

Reply:
Thanks